external-crates/move/crates/move-compiler/src/expansion/ast.rs

```diff
@@ -261,6 +261,13 @@ pub type Type = Spanned<Type_>;
 // Expressions
 //**************************************************************************************************
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum Mutability {
+    Imm,
+    Mut(Loc), // if the local had a `mut` prefix
+    Either,   // for legacy and temps
```

Comment:
FWIW I think temps are subject to immutability. Match temps in the enum PR may be Either, though.